wdtk-wikibaseapi/src/main/java/org/wikidata/wdtk/wikibaseapi/ApiConnection.java

```diff
@@ -335,6 +345,7 @@ public void logout() throws IOException {
 			Map<String, String> params = new HashMap<>();
 			params.put("action", "logout");
 			params.put("format", "json"); // reduce the output
+			params.put("token", fetchToken("csrf"));
```

Comment:
Maybe`getOrFetchToken` if the csrf token has been already loaded?

Reply:
Yes, I probably missed that as I originally tested with this code.
One more thing to consider might the re-fresh of CSRF certificate as in ```WBEditAction```, as I'm not sure whether the close may fail with "older"? CSRF certificate.